Dockerfile

```diff
@@ -1,45 +1,41 @@
-# build, lint, and test
+# lint, test, transpile and buid binary (as deb)
 FROM node:12.4 as builder
 WORKDIR /usr/app/builder
 COPY package*.json ts*.json ./
 COPY ./src ./src
+# use npm to complete lifecycle operations
 RUN npm install
 RUN npm run lint
-RUN npm run build
 RUN npm run test
+RUN npm run dist
 
-FROM node:12.4 as runner
-# Get all the x/os deps we need
-RUN apt-get update \
-  && DEBIAN_FRONTEND=noninteractive apt-get install -y curl \
-  dos2unix \
-  xinit \
-  libnss3 \
-  libxss1 \
-  libasound2 \
-  libdbus-1-3 \
-  libgdk-pixbuf2.0-0 \
-  libgtk-3-0 \
-  xserver-xorg-video-dummy \
-  x11-xserver-utils \
-  cnee \
-  fonts-liberation \
-  fonts-roboto
+# setup runtime
+FROM debian:stretch-slim as runner
 # Create a lower-priv user (electron) to run as
 RUN groupadd -r electron \
-  && useradd -r -g electron -G audio,video,tty electron \
+  && useradd -r -g electron -G audio,video,tty,dialout electron \
   && mkdir -p /home/electron/Downloads \
   && chown -R electron:electron /home/electron \
-  && mkdir /home/electron/.config
+  && mkdir /home/electron/.config \
+  && chown -R electron:electron /home/electron/.config
 WORKDIR /home/electron/browserd
+COPY --from=builder /usr/app/builder/bin/*.deb ./
+# install the deb
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -y \
+  && apt-get install -y xvfb xauth libgl1-mesa-dri \
+  && apt-get install -y ./*.deb \
+  && rm -rf /var/lib/apt/lists/*
 
+# build release image
 FROM runner as release
-# move in resources
-COPY --from=builder /usr/app/builder/package*.json ./
-COPY --from=builder /usr/app/builder/dist ./dist
-RUN npm ci
-COPY .docker-display.conf .
-COPY .docker-x-boot.sh .
-RUN dos2unix .docker-display.conf .docker-x-boot.sh
-RUN chmod +x .docker-x-boot.sh
-CMD /home/electron/browserd/.docker-x-boot.sh 'npm run direct-start -- --no-sandbox'
+# we start as root, but quickly drop to "electron"
+USER root
+# as root:
+# start dbus system
+# as electron:
+# start dbus session
+# start x11 session (xvfb)
+# start browserd
+CMD service dbus start \
+  && su electron -c 'dbus-run-session -- xvfb-run -s "-ac -br -nocursor -screen 0 1920x1080x24 -nolisten tcp" browserd --no-sandbox --disable-gpu'
```

Comment:
Ah, so we've changed the resolution to 1920x1080.

Reply:
Should we read these values from `.env`?

Reply:
no. this is build-time config, not run-time config. its up to the user to manually configure this if they want other values.

Reply:
Then I'd love this resolution settings is mentioned somewhere in our docs.